### PR TITLE
Issue #1030 c8y_configuration_plugin integration tests

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

Integration tests for c8y_configuration_plugin

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by rina23q — COMMENTED** [review]
I see only minor fixes are needed.

File: crates/core/c8y_api/src/http_proxy.rs [hunk 1]
```diff
@@ -47,7 +47,6 @@ pub trait C8YHttpProxy: Send + Sync {
     async fn upload_config_file(
         &mut self,
         config_path: &Path,
-        config_content: &str,
```
[the remark this answers is not in the document]
Comment (albinsuresh) [reply crates/core/c8y_api/src/http_proxy.rs:50]: Yes, it was done to make the testing easier.

For your usecase to support `config_type`, I'd just suggest that you add it as an extra `Option<String>` argument in addition to the `config_path: &Path` argument. When this `config_type` is `None`, we can internally derive the default value from `config_path`.

**Review by rina23q — APPROVED** [review]

File: plugins/c8y_configuration_plugin/src/main.rs [hunk 2]
```diff
@@ -185,3 +194,78 @@ fn create_operation_files(config_dir: &str) -> Result<(), anyhow::Error> {
...
+            &mut messages,
+            TEST_TIMEOUT_MS,
+            &[
+                "501,c8y_UploadConfigFile",
+                "503,c8y_UploadConfigFile,http://server/some/test/config/url",
+            ],
+        )
+        .await;
+
+        Ok(())
```
Comment (initard) [inline plugins/c8y_configuration_plugin/src/main.rs:269]: very nice test

**Review by initard — APPROVED** [review]
LGTM

**Pull request merged**